### ISSUE TITLE
Document Windows setup for local development

### ISSUE THREAD

**Issue by maestromac** [thepracticaldev/dev.to#306]
@nickytonline commented on [Wed Aug 08 2018](https://github.com/thepracticaldev/dev.to_core/issues/674)

# _TASK_
As soon as the code is open source, this is going to become a priority.

Document local environment setup for Windows and modify any scripts (if necessary) to get a development environment up and running for Windows.

If you're on a Mac, install a [free VM](
https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/) from Microsoft.

### Definition of Done
Documentation and any script changes (if any) required setting up a development environment on Windows.

---

@Zhao-Andy commented on [Wed Aug 08 2018](https://github.com/thepracticaldev/dev.to_core/issues/674#issuecomment-411464473)

@nickytonline Do you have any experience with setting up for Windows? If so, how does this guide look? https://gorails.com/setup/windows/10 I referenced it since that's the guide I use for setting up Rails on macOS.

Think I'll give it a try on my Windows laptop when I have some time.

Edit: added link oops https://gorails.com/setup/windows/10

---

@nickytonline commented on [Wed Aug 08 2018](https://github.com/thepracticaldev/dev.to_core/issues/674#issuecomment-411606074)

Hey @Zhao-Andy. I created this issue because I read somewhere in the docs that [all dev by the core team was currently done on Macs only](
https://docs.dev.to/faq/#how-do-i-setup-the-repo-with-windows/linux/ubuntu/not-macos?). I have never installed Ruby on Windows, just node.js. For Node.js, the updates I made to the front-end doc should cover Windows as well aside from the option to install via `brew`.

I haven't checked out the rails doc you mentioned yet. I think it would be great to just have the whole local dev setup officially documented for Windows and Linux. This would  make a great first PR to the repo for documentation.

---

@joshi-chinmay commented on [Wed Aug 08 2018](https://github.com/thepracticaldev/dev.to_core/issues/674#issuecomment-411623938)

Hey, @Zhao-Andy and @nickytonline, I have a `Windows 8.1` machine. But I have never done any development setup on it. I would be more than happy to tweak around the Windows environment and write a document about it. What say?

---

@Zhao-Andy commented on [Thu Aug 09 2018](https://github.com/thepracticaldev/dev.to_core/issues/674#issuecomment-411742088)

Sure, go for it! That'll be super helpful, and I'll compare notes with you when I run my Windows 10 machine with the GoRails guide. The GoRails guide only has a guide available for Windows 10, so I'm not sure if it'll work for 8.1.

Link: https://gorails.com/setup/windows/10

---

@rhymes commented on [Thu Aug 09 2018](https://github.com/thepracticaldev/dev.to_core/issues/674#issuecomment-411761683)

@joshi-chinmay you can download free and temporary VMs from Microsoft's website if you need one https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/

**Comment by nathansebhastian** [comment]
Hi everyone,

 I'd like to help contribute to dev.to, and since I have windows machine for development with WSL, I think I'll start here. I've followed the documentation and gorails tutorials, right now the site is running fine on my localhost. Here are the outline of the steps I did to run it:

1. Install WSL 18.04 and check for postgre first, use apt install 

1. Install NodeJS 8 from debsource
    ```
    curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
    sudo apt-get install -y nodejs
    ```
    If npm -v give Syntax error: word unexpected (expecting "in"), you can try restart the terminal. It works for me

1. Install yarn from debsource
[https://yarnpkg.com/en/docs/install#debian-stable](https://yarnpkg.com/en/docs/install#debian-stable)

1. Install ruby 2.5.1 using gorails tutorial

1. skip git configuration because I have one already

1. Install Rails 5.1.6 using gorails tutorial

1. Fork dev.to git and clone it to my local machine

1. gem install bundler

1. gem install foreman

1. run `rbenv rehash`

1. bundle install

1. bin/yarn --network-timeout 1000000 (I encountered this [issue](https://github.com/yarnpkg/yarn/issues/4890))

1. Get API keys (algolia only)

1. Configure database.yaml for to connect to postgresql run in Windows instead of Ubuntu. make sure it looks like the following
    ```
    development:
    <<: *default
    database: PracticalDeveloper_development
    host: localhost
    username: postgres
    password: root
    ```
    you have to add **host: localhost** else Rails will look to connect to Unix Postgres and not Windows.
Postgres in WSL still have some issues. Look at this [issue](https://github.com/Microsoft/WSL/issues/2628)

1. if statement timeout, comment the variable out

1. bin/setup and wait until finished

1. bin/startup. Running jobs take a while the first time.

1. Open your Windows Chrome and hit localhost:3000

![dev to running on windows](https://user-images.githubusercontent.com/10413648/45261385-c2862f00-b42a-11e8-9eba-38f230426bcf.png)

![dev to running on windows 2](https://user-images.githubusercontent.com/10413648/45261389-d03bb480-b42a-11e8-9d80-13bd64a4a342.png)

I haven't don't anything except browsing around, will include other API keys and try to sign up later. I hope this helps.

**Comment by nathansebhastian** [comment]
UPDATE: I've installed Ruby and rails with the exact specific version on Windows. Here are some errors I found:

- Rails complain sys-proctable gem were not found in the gemfile. Solved with `bundle add sys-proctable`

- Foreman `spawn': Exec format error - bin/rails s -p 3000 (Errno::ENOEXEC) 
    Edit the procfile.dev file into 
    ```
    web: ruby bin/rails s -p 3000
    webpacker: ruby ./bin/webpack-dev-server
    job: ruby bin/rake jobs:work
    ```
- Webpacker ./bin/webpack-dev-server:65:in `exec': No such file or directory
    I didn't find the solution for this, and unfortunately my time is up for tweaking things.

Guess I'll go with the WSL setup for now

**Comment by maestromac** [comment]
Hey @nathansebhastian , I really like what you shared here! feel to create PR to update [this file](https://github.com/thepracticaldev/dev.to/blob/master/docs/installation/windows.md). If you are uncertain on how you want to format it, mimicing [our linux guide](https://github.com/thepracticaldev/dev.to/blob/master/docs/installation/linux.md) is a good call.

**Comment by nathansebhastian** [comment]
Hi @maestromac okay then I'll make one when it's ready for review

**Comment by Nirajn2311** [comment]
@maestromac did you install postgre in windows or in wsl

**Comment by maestromac** [comment]
@Nirajn2311 I haven't attempted to install postgres on my windows machine yet. I'll try it out and let you know.

**Comment by nathansebhastian** [comment]
@Nirajn2311 I've installed on both, but for WSL I encountered an issue which I mentioned in step 14. For now my local build in WSL is connecting to Windows PSQL

**Comment by Nirajn2311** [comment]
@nathansebhastian When I was installing the pg gem it ran into an error due to missing libraries. The solution to fix this issue was downloading the libraries with the command `sudo apt-get install libpq-dev`

Also the database username and password also have to be manually added to test database in database.yml

The above two were the only issues I ran into while setting up.

**Comment by nathansebhastian** [comment]
@Nirajn2311 Ah now that you mention it, yes I do have that pg gem error on install.. Noted to add these two points in the documentation.. Thanks!

**Comment by rhymes** [comment]
@Nirajn2311 @nathansebhastian this is how you can avoid editing database.yml to add username and password https://docs.dev.to/additional-postgres-setup/#configuration

**Comment by nathansebhastian** [comment]
Hi everyone, I've created the PR to update Windows documentation. Please check it out.

I have added both @Nirajn2311 and @rhymes comment into it as well.

**Comment by maestromac** [comment]
Oh, this was resolved with #671
